### PR TITLE
New TradingHours EA

### PR DESCRIPTION
**Pull request by martin-cll**
This PR implements a new TradingHours external adapter with the new MarketStatus endpoint. See [Requirements doc](https://docs.google.com/document/d/1z96Rrj0TDp6hwTlMidyPc8gOQqbbyit-5nm81SJ3zW8/edit#heading=h.gjdgxs).

High level details:
* Env vars: Optional endpoint URL and required API key
* Parameters: `market` name, one of `forex`, `metals`, or `wti`
* Integration type: HTTP
* Rate limits: 1 req/s
* Sends bulk requests (i.e. multiple markets per req) per tick to their `/v3/markets/status` API endpoint.
  * Our `market` param is translated to their `fin_id` identifiers.

Output:
```
➜  external-adapters-js git:(ml/tradinghours-ea) echo '{"data":{"endpoint":"market-status", "market":"wti"}}' | http -v POST localhost:8080 Content-Type:application/json
POST / HTTP/1.1
Accept: application/json, */*;q=0.5
Accept-Encoding: gzip, deflate
Connection: keep-alive
Content-Length: 54
Content-Type: application/json
Host: localhost:8080
User-Agent: HTTPie/3.2.3

{
    "data": {
        "endpoint": "market-status",
        "market": "wti"
    }
}


HTTP/1.1 200 OK
Connection: keep-alive
Date: Sun, 11 Aug 2024 21:56:33 GMT
Keep-Alive: timeout=72
content-length: 277
content-type: application/json; charset=utf-8

{
    "data": {
        "result": 1
    },
    "meta": {
        "adapterName": "TRADINGHOURS",
        "metrics": {
            "feedId": "{\"market\":\"wti\"}"
        }
    },
    "result": 1,
    "statusCode": 200,
    "timestamps": {
        "providerDataReceivedUnixMs": 1723413393056,
        "providerDataRequestedUnixMs": 1723413392920,
        "providerIndicatedTimeUnixMs": 1723413394000
    }
}
```

Note: Adapter-secrets PR is pending access for my github account.

### CODE REVIEW

File: packages/sources/tradinghours/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+| Required? |     Name     |            Description            |  Type  | Options |            Default             |
+| :-------: | :----------: | :-------------------------------: | :----: | :-----: | :----------------------------: |
+|           | API_ENDPOINT | Base URL for the TradingHours API | string |         | `https://api.tradinghours.com` |
+|    ✅     |   API_KEY    | API key for the TradingHours API  | string |         |                                |
+
+---
+
+## Data Provider Rate Limits
+
+There are no rate limits for this adapter.
```
Comment (mjk90) [inline packages/sources/tradinghours/README.md:18]: Should this be updated with the 1 req/s mentioned in the PR?

File: packages/sources/tradinghours/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# TRADINGHOURS
+
+![0.1.0](https://img.shields.io/github/package-json/v/smartcontractkit/external-adapters-js?filename=packages/sources/tradinghours/package.json) ![v3](https://img.shields.io/badge/framework%20version-v3-blueviolet)
```
Comment (mjk90) [inline packages/sources/tradinghours/README.md:3]: This version doesn't seem to match the version in package.json. Was this generated automatically?

**Review by mjk90 — COMMENTED** [review]
Looks good to me, just a couple of notes on the readme 🙏

File: packages/sources/tradinghours/README.md (hunk 2 above)
Comment (martin-cll) [reply packages/sources/tradinghours/README.md:3]: Yup the readme was outdated, just re-generated it!

File: packages/sources/tradinghours/package.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  "scripts": {
+    "clean": "rm -rf dist && rm -f tsconfig.tsbuildinfo",
+    "prepack": "yarn build",
+    "build": "tsc -b",
+    "server": "node -e 'require(\"./index.js\").server()'",
+    "server:dist": "node -e 'require(\"./dist/index.js\").server()'",
+    "start": "yarn server:dist"
+  },
+  "dependencies": {
+    "@chainlink/external-adapter-framework": "1.3.0",
```
Comment (mxiao-cll) [inline packages/sources/tradinghours/package.json:31]: Can we use the latest version?

**Review by mxiao-cll — APPROVED** [review]

File: packages/sources/tradinghours/package.json (hunk 3 above)
Comment (martin-cll) [reply packages/sources/tradinghours/package.json:31]: Yup, this uses the same version that all other EAs are currently using. I can make a separate PR to update all EAs.

**Pull request merged**